src/exo/API_scheduling.py

```diff
@@ -2199,6 +2537,15 @@ def specialize(proc, block_cursor, conds):
         block_cursor    - cursor pointing to the block to duplicate/specialize
         conds           - list of strings or string to be parsed into
                           guard conditions for the
+        rc              - bool (whether to return relevant cursors)
+
+    relevant_cursors:
+        if              - a cursor to the outermost if-statement
+        TODO: should we return each of the subcases? There's a natural navigation
```

Comment:
Some scheduling operations seem like they should maybe be split into multiple scheduling operations. There's `specialize`, `lift_scope` and `eliminate_dead_code`

Reply:
I don't understand why `specialize` should be split into something more primitive.  I think either of the options of "just return the one cursor" or "return the one outer cursor and also return a tuple of cursors, one for the body of each branch" are both reasonable specs for `rc` here.